flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionWithReadViewTest.java

```diff
@@ -353,14 +354,32 @@ private void testBacklogConsistentWithNumberOfConsumableBuffers(boolean isFlushR
 		}
 	}
 
+	@Test
+	public void testBlockedByCheckpointAndResumeConsumption() throws IOException, InterruptedException {
+		subpartition.add(createFilledFinishedBufferConsumer(BUFFER_SIZE));
+		subpartition.add(createEventBufferConsumer(BUFFER_SIZE, Buffer.DataType.EXACTLY_ONCE_CHECKPOINT_BARRIER));
+		subpartition.add(createEventBufferConsumer(BUFFER_SIZE, Buffer.DataType.EVENT_BUFFER));
+		subpartition.add(createFilledFinishedBufferConsumer(BUFFER_SIZE));
+		subpartition.finish();
```

Comment:
it is no need to call `finish` for this test, otherwise we also need to assert `EndofPartitionEvent` below.